src/Orleans.Runtime/Catalog/Catalog.cs

```diff
@@ -443,20 +439,19 @@ public void GetGrainTypeInfo(int typeCode, out string grainClass, out PlacementS
             activatedPromise = Task.CompletedTask;
             PlacementStrategy placement;
 
-            lock (activations)
+            while (true)
```

Comment:
this smells bad. why are we adding infinite retry here while the original logic didn't?

Reply:
this is a standard lock-free approach when dealing with optimistic concurrency conflicts

---
In reply to: [274565380](https://github.com/dotnet/orleans/pull/5526#discussion_r274565380) [](ancestors = 274565380)

Reply:
I see. Thanks for explaining !